modules/manage/pages/kubernetes/k-manage-resources.adoc

```diff
@@ -7,27 +7,55 @@
 
 You can define requirements for Pod resources such as CPU, memory, and storage. Redpanda Data recommends that you determine and set these values before deploying the cluster, but you can also update the values on a running cluster.
 
-NOTE: In a running Redpanda cluster, you cannot decrease the number of CPU cores. You can only increase the number of CPU cores.
+== Prerequisites
 
-To see the available resources on the worker nodes that you have provisioned:
+- See xref:deploy:deployment-option/self-hosted/kubernetes/k-requirements.adoc[] for the minimum worker node, memory, CPU, and storage requirements.
 
+- Make sure that your physical or virtual machines have enough resources to give to Redpanda. To see the available resources on the worker nodes that you have provisioned:
++
 [,bash]
 ----
 kubectl describe nodes
 ----
 
-== Prerequisites
+== Production considerations
+
+- <<cpu, Enable CPU pinning>>. This configuration allows Redpanda to maintain exclusive access to designated CPU cores, improving throughput and reducing latency caused by CPU context switches.
+
+- <<qos, Enable the `Guaranteed` quality of service class for Pods that run Redpanda>>. This setup ensures that the CPU and memory allocated to Redpanda are not subject to throttling or other contention issues, providing a stable and predictable performance environment.
+
+- <<memory, Enable memory locking>>. This configuration prevents the operating system from paging out Redpanda's memory to disk, which can significantly impact performance.
+
+== Limitations
 
-See xref:deploy:deployment-option/self-hosted/kubernetes/k-requirements.adoc[] for memory, CPU, and storage.
+You cannot decrease the number of CPU cores in a running cluster.
 
-Make sure that your physical or virtual machines have enough resources to give to Redpanda. You assign resources to each Redpanda broker using the `values.yaml` settings. If you need to replace a worker node with one that has more resources, xref:./decommission-brokers.adoc[decommission the brokers] before you replace the worker nodes.
+[[cpu]]
+== CPU pinning
 
-== Configure CPU resources
+Redpanda is designed to maximize throughput by xref:get-started:architecture.adoc#tpc[pinning its threads to specific CPU cores]. However, CPU pinning is disabled by default in Kubernetes because it uses the https://www.kernel.org/doc/html/latest/scheduler/sched-design-CFS.html[Completely Fair Scheduler (CFS)] to dynamically manage CPU allocation. The CFS distributes CPU time among all processes based on their priority and resource requests, which can lead to frequent context switches and migration of processes across different cores. This behavior can degrade the performance of Redpanda due to:
```

Comment:
It's not explicitly disabled by anything in the Kubernetes layer. I am curious to know if seastar is doing any type of detection. How about saying "CPU pinning is, by default, unavailable in Kubernetes due to its usage of the CFS"?

It may also be useful to introduce the term "hardware locality here"? Another way to explain the performance impact could be something like: "Redpanda and Seastar rely on hardware locality for optimal performance. CFS is unaware of hardware locality and can result in a negative performance impact due to:"

Reply:
Mentioned in the bulleted list below this paragraph.